Sources/StreamChat/Database/DTOs/AttachmentDTO.swift

```diff
@@ -244,7 +246,16 @@ extension AttachmentDTO {
             }
             image.imageURL = uploadedFileURL
             data = try? JSONEncoder.stream.encode(image)
-        case .file:
+        case .video:
+            guard var video: VideoAttachmentPayload = payload() else {
+                log.assertionFailure(
+                    "Video payload must be decoded to provide the `videoURL` before sending"
+                )
+                return
+            }
+            video.videoURL = uploadedFileURL
+            data = try? JSONEncoder.stream.encode(video)
+        default:
```

Comment:
Why are we defaulting to file?

Reply:
`AnyAttachmentPayload` can be created with local URL and random attachment type. It's some kind of `custom uploadable attachments` feature (no other fields can be added/changed but the `type` though). 

To complete uploading of such kind of attachments we have a `default` case (we don't know the exact type but we already know that's not one of the built-in uploadable types). It's safe since non-uploadable won't be picked up by the uploader so we'll never get there.

Reply:
While writing I found that there's no way to get such attachments from the message because the type check will fail 🤔 

```swift
func attachment<Payload: AttachmentPayload>(payloadType: Payload.Type) -> _ChatMessageAttachment<Payload>? {
 guard Payload.type == type || type == .unknown else { return nil }
 ...  
}
```

@VojtaStavik should we find the way to expose them or start trowing an error when creating a payload? 🤔

Reply:
@VojtaStavik WDYT?

Reply:
Updated to throw an error when custom type is given. This feature was half done and such attachments were not accessible on the message. The full support of custom uploadable attachments (when both a file and custom payload can be attached) would require more effort anyway.